java/integTest/src/test/java/glide/TestUtilities.java

```diff
@@ -76,4 +80,51 @@ public static Map<String, String> parseInfoResponseToMap(String serverInfo) {
         return RedisClusterClientConfiguration.builder()
                 .address(NodeAddress.builder().port(CLUSTER_PORTS[0]).build());
     }
+
+    /**
+     * Deep traverse and compare two objects, including comparing content of all nested collections
+     * recursively.
+     *
+     * @apiNote Maps and Set comparison ignores their order, regardless of `orderMatters` argument.
+     *     Their entries could be reordered, but values stored in them compared according to this
+     *     parameter.
+     */
+    public static void assertDeepEquals(Object expected, Object actual, boolean orderMatters) {
+        if (expected == null || actual == null) {
+            assertEquals(expected, actual);
```

Comment:
isn't `null == null` always going to be true? could there be different references to null values in java?

Reply:
There is a case, when one is null, but second isn't - we need to catch it early, otherwise we have NPE on the next line.